src/Controller/BackendCsvImportController.php

```diff
@@ -0,0 +1,250 @@
+     *
+     * @param DataContainer $dc
+     *
+     * @return string
+     *
+     * @throws RedirectResponseException
+     */
+    public function importOptionWizard(DataContainer $dc)
+    {
+        if (($csvImport = $this->createImportInstance()) === null) {
+            return '';
+        }
+
+        $csvImport->setCallback(function (array $data, array $row) {
+            $data[] = [
+                'value' => $row[0],
+                'label' => $row[1],
+            ];
+
+            return $data;
+        });
+
+        return $this->runDefaultRoutine($csvImport, $dc, 'options', $GLOBALS['TL_LANG']['MSC']['ow_import'][0]);
+    }
+
+    /**
+     * Generate the import view for table wizard
+     *
+     * @param DataContainer $dc
+     *
+     * @return string
+     *
+     * @throws RedirectResponseException
+     */
+    public function importTableWizard(DataContainer $dc)
+    {
+        if (($csvImport = $this->createImportInstance()) === null) {
+            return '';
+        }
+
+        $csvImport->setCallback(function (array $data, array $row) {
+            $data[] = $row;
+
+            return $data;
+        });
+
+        $csvImport->setSeparators([
+            CsvImportUtil::SEPARATOR_COMMA,
+            CsvImportUtil::SEPARATOR_SEMICOLON,
+            CsvImportUtil::SEPARATOR_TABULATOR,
+        ]);
+
+        return $this->runDefaultRoutine($csvImport, $dc, 'tableitems', $GLOBALS['TL_LANG']['MSC']['tw_import'][0]);
+    }
+
+    /**
+     * Create the CSV import instance
+     *
+     * @return CsvImportUtil|null
+     */
+    private function createImportInstance()
+    {
+        if (($uploader = $this->getDefaultUploader()) === null) {
+            return null;
+        }
+
+        return new CsvImportUtil(
+            $this->connection,
+            $this->flashBag,
+            $this->requestStack->getCurrentRequest(),
+            $uploader
+        );
+    }
+
+    /**
+     * Run the default import routine
+     *
+     * @param CsvImportUtil $csvImport
+     * @param DataContainer $dc
+     * @param string        $fieldName
+     * @param string        $submitLabel
+     *
+     * @return string
+     *
+     * @throws RedirectResponseException
+     */
+    private function runDefaultRoutine(CsvImportUtil $csvImport, DataContainer $dc, $fieldName, $submitLabel)
+    {
+        if ($csvImport->isFormSubmitted()) {
+            try {
+                $csvImport->run($dc->table, $fieldName, $dc->id);
+            } catch (CsvImportErrorException $e) {
+                // Add an error message and reload the page on import failure
+                $this->flashBag->add('error', $e->getMessage());
+                throw new RedirectResponseException($this->requestStack->getCurrentRequest()->getRequestUri());
+            }
+
+            // Set the backend offset cookie
+            $this->framework->getAdapter('Contao\System')->setCookie('BE_PAGE_OFFSET', 0, 0);
```

Comment:
I wonder if we really have to reset the Cookie.
If not, we only need to inject the max filesize below and are completely independent from the framework in this class, which I consider a goal for every new class in Contao.